draft-meynell-panrg-scion-research-questions.md

```diff
@@ -200,6 +200,26 @@ This draft assumes the reader is familiar with some of the fundamental concepts
 * One AS belongs to many ISDs?
 
 
+## CORE Beaconing Scalability
+
+The idea behind beaconing is to discover all possible paths (loop free and with a fixed maximu length) between two CORE AS.
+Every AS forwards any received becons to all neighbor CORE ASes unless this would cause a loop or exceed the fixed maximum length.
+Implemented naively, the number of paths (and beacons) grows exponentially with the network size.
+This is currently mitigated primarily (and efficiently) by forwarding only the 5 "best PCB" to neighbours.
+
+This "best PCB" practive has several implications:
+
+* It limits the number of available paths for endhosts such that only 5 path may be available to a given destination. This is especially relevant for
+  * Multipathing: you may not be able to use more than five path and no great selection of pats on offer .
+  * Unusual path policies, such as geofencing, may not be fulfillable by the limited number of paths on offer.
```

Comment:
```
  * Unusual path policies, such as geofencing, may not be fulfillable by the limited number of paths on offer.
```
Was this ever an actual problem? Is it an edge case? I've never heard of this before

Reply:
Not sure it happened yet, simply because we do not have much useful metadata yet (geofencing etc).
I think it can become a problem: if all ASes only forward their best five paths, then these five paths may be disjunct in themselves, but neighboring ASes are probably likely to forward the (almost) same 5 paths. In effect, an endpoint may have more than 5 paths available, but they are likely to be very similar and may not offer the choice that an endpoint desires?

Reply:
I agree with @tzaeschke _iff_ they forward the same PCBs during each beaconing period.
Let me give an example to clarify:
- First AS "A" discriminates good PCBs using latency criteria. This means that for AS "A" only good latency PCBs are forwarded.
- Second AS "B" receives PCBs from AS "A", but "B" discriminates bases on bandwidth. Maybe some PCBs have a good bandwidth, but many were already filtered by "A".
- etc.

If ASes forward _all received beacons_ (in general, and assuming there is no expiration time inside the PCBs), there should not be a problem, since all paths will be eventually discovered.

Now, because reality is that ASes cannot forward all of the PCBs, the non forwarded PCBs (those that an AS never forwards) are the least interesting ones _according to the local criteria of that AS_, which may or may not coincide with that criteria of the next neighbors this PCB will encounter.

I believe this to be clearer if explained in terms of completeness of the graph exploration vs. time needed for it.